lib/request.ts

```diff
@@ -119,89 +130,114 @@ export const request = (
         originalReject(errorWithTimings);
       };
     }
-    const requestObject = httpRequestFn(
-      options,
-      (response: IncomingMessage) => {
+
+    const requestObject = httpRequestFn(options);
+
+    function setReadTimeout() {
+      requestObject.setTimeout(readTimeout, () => {
+        requestObject.socket.destroy();
+        reject(
+          new Error(
```

Comment:
Might be nice to make these custom error types as well to follow current pattern